sycl/source/device_selector.cpp

```diff
@@ -192,6 +222,13 @@ int filter_selector::operator()(const device &Dev) const {
 
 void filter_selector::reset() const { impl->reset(); }
 
+// filter_selectors not "Callable"
+// because of the requirement that the filter_selector "reset()" itself
+// between invocations, the filter_selector operator() is not purely callable
+// and cannot be used interchangeably as a SYCL2020 callable device selector.
+// TODO: remove this reset() requirement
+// and remove ! std::is_base_of_v<ext::oneapi::filter_selector, DeviceSelector>
+// from device/platform/queue constructors
```

Comment:
Ah, this is an important & interesting design information that appears late in the reading of this PR. :-)
Yes, it would be nice to have another API for your `filter_selector`.

Reply:
I have to update this comment. The plan is to replace the FilterSelector class with something that doesn't pretend to be a device_selector, and instead is something that just returns a device (rather than a score).